File: content/realm/examples/dotnet/Examples/Plant.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using Realms;

namespace Examples.Models
{
    // :snippet-start: plant-class
    public partial class Plant : IRealmObject
    {
        //:remove-start:
        [PrimaryKey]
        [MapTo("_id")]
        //:remove-end:
        [BsonElement("_id")]
        public ObjectId Id { get; set; } = ObjectId.GenerateNewId();

        [BsonElement("name")]
        public string? Name { get; set; }

        [BsonElement("sunlight")]
        [BsonRepresentation(BsonType.String)]
        public string? Sunlight { get; set; }

        [BsonElement("color")]
        [BsonRepresentation(BsonType.String)]
        public string? Color { get; set; }

        [BsonElement("type")]
        [BsonRepresentation(BsonType.String)]
        public string? Type { get; set; }

        [BsonElement("_partition")]
        public string? Partition { get; set; }
    }
    public enum Sunlight
    {
        Full,
        Partial
    }
    public enum PlantColor
    {
        White,
        Green,
        Yellow,
        Purple
    }
    public enum PlantType
    {
        Perennial,
        Annual
    }
    // :snippet-end:
}

File: content/realm/examples/java/sync/app/src/main/java/com/mongodb/realm/examples/model/Plant.kt
package com.mongodb.realm.examples.model
// :snippet-start: plant
import org.bson.codecs.pojo.annotations.BsonProperty
import org.bson.types.ObjectId

open class Plant(val id : ObjectId = ObjectId(),
                 var name : String? = null,
                 var sunlight : String? = null,
                 var color : String? = null,
                 var type : String? = null,
                 @field:BsonProperty("_partition") // specify that this is a field-level annotation
                 var partition : String? = null) {
    override fun toString(): String {
        return "Plant [id=$id, name=$name, sunlight=$sunlight, color=$color, type=$type, partition=$partition]"
    }
}
// :snippet-end: